### PR TITLE
added hsv parameter to ColorArray

### LINKED ISSUE #692: Add hsv option in ColorArray's constructor
```python
ColorArray(hsv=mynx3array)
```

### PR DESCRIPTION
**Pull request by bollu**
implemented the [feature request](https://github.com/vispy/vispy/issues?q=hsvis%3Aissue+is%3Aopen+) to allow hsv parameters to ```ColorArray```

closes #692

### CODE REVIEW

File: vispy/color/color_array.py [hunk 1]
```diff
@@ -124,9 +128,24 @@ class ColorArray(object):
     Under the hood, this class stores data in RGBA format suitable for use
     on the GPU.
     """
-    def __init__(self, color='black', alpha=None, clip=False):
+    def __init__(self, color=None, color_space='rgb',
+                 alpha=None, clip=False):
```
Comment (Eric89GXL) [inline vispy/color/color_array.py]: be careful about backward compat, the new arg should go at the end not be inserted earlier

**Comment by Eric89GXL** [comment]
Other than my minor comment, LGTM (and thanks for adding tests!)

File: vispy/color/color_array.py (hunk 1 above)
Comment (bollu) [inline vispy/color/color_array.py]: ah, thanks :)
Had never thought about that.

**Comment by bollu** [comment]
build passed :)

File: vispy/color/color_array.py [hunk 2]
```diff
@@ -91,6 +91,10 @@ class ColorArray(object):
         If array-like, it must be an Nx3 or Nx4 array-like object.
         Can also be a list of colors, such as
         ``['red', '#00ff00', ColorArray('blue')]``.
+    color_space : "rgb" | "hsv"
+       if rgb, color tuples are interpreted as (r, g, b) components.
+       if hsv, color tuples are interpreted as (h, s, v) components.
+       if neither are specified, it is assumed to "rgb" by default
```
Comment (QuLogic) [inline vispy/color/color_array.py]: s/to/to be/

File: vispy/color/color_array.py [hunk 3]
```diff
@@ -124,9 +128,24 @@ class ColorArray(object):
     Under the hood, this class stores data in RGBA format suitable for use
     on the GPU.
     """
-    def __init__(self, color='black', alpha=None, clip=False):
+    def __init__(self, color=None, alpha=None,
+                 clip=False, color_space='rgb'):
+
+        # if color is RGB, then set the default color to black
```
Comment (QuLogic) [inline vispy/color/color_array.py]: This comment is the same as the one below.

File: vispy/color/color_array.py [hunk 4]
```diff
@@ -124,9 +128,24 @@ class ColorArray(object):
...
+            color = 'black' if color is None else color
+        elif color_space == 'hsv':
+            color = (0, 0, 0) if color is None else color
+            # if the color space is hsv, convert hsv to rgb
+            color = _hsv_to_rgb(color)
+        else:
+            raise ValueError('color_space should be either "rgb" or'
+                             '"hsv", it is ' + color_space)
+
         """Parse input type, and set attribute"""
```
Comment (QuLogic) [inline vispy/color/color_array.py]: This line is odd. As a doc string, you might think it should stay at the top of the function, but there's already a doc string before the function. So maybe it should be a comment? (This is not a question about this PR specifically, BTW.)

**Comment by QuLogic** [comment]
You might want to add an example to the docstring just above `__init__`, too?

File: vispy/color/color_array.py [hunk 5]
```diff
@@ -91,6 +91,10 @@ class ColorArray(object):
         If array-like, it must be an Nx3 or Nx4 array-like object.
         Can also be a list of colors, such as
         ``['red', '#00ff00', ColorArray('blue')]``.
+    color_space : "rgb" | "hsv"
+       if rgb, color tuples are interpreted as (r, g, b) components.
+       if hsv, color tuples are interpreted as (h, s, v) components.
+       if neither are specified, it is assumed to be "rgb" by default
```
Comment (Eric89GXL) [inline vispy/color/color_array.py]: You can just say `(default : "rgb")`. Also, please capitalize the `if`'s at the start of the sentences.

**Comment by Eric89GXL** [comment]
Other than my nitpick LGTM, let me know when it's fixed and the CIs are happy and I'll merge.

**Comment by Eric89GXL** [comment]
Thanks for another fix!

**Pull request merged**